dbms/src/Storages/StorageDeltaMerge.cpp

```diff
@@ -767,6 +767,15 @@ DM::PushDownFilterPtr StorageDeltaMerge::parsePushDownFilter(const SelectQueryIn
         const auto & table_infos = tidb_table_info.columns;
         for (const auto & col : columns_to_read)
         {
+            // table_infos does not contain handle column
+            if (col.id == -1)
+            {
+                auto handle = ColumnInfo();
+                handle.id = EXTRA_HANDLE_COLUMN_ID;
+                handle.name = EXTRA_HANDLE_COLUMN_NAME;
+                table_scan_column_info.push_back(handle);
+                continue;
+            }
             auto iter = std::find_if(
```

Comment:
Better show the `col.id` and `col.name` in L783  in case similar problem happen again
https://github.com/pingcap/tiflash/blob/405102e80f89f5daddab0023d075bc02e91f48c8/dbms/src/Storages/StorageDeltaMerge.cpp#L779-L783

Reply:
And also L761

Reply:
> Better show the `col.id` and `col.name` in L783 in case similar problem happen again

done

> And also L761

`columns_to_read` will contain extre column, so there is no need to check in L761